cowbird/utils.py

```diff
@@ -155,20 +163,58 @@ def islambda(func):
     return isinstance(func, types.LambdaType) and func.__name__ == (lambda: None).__name__  # noqa
 
 
-def get_app_config(container, celery=True):
+def configure_celery(config, config_ini):
+    logger = get_logger(__name__)
+    logger.info("Configuring celery")
+
+    # shared_tasks use the default celery app by default so setting the pyramid_celery celery_app as default prevent
+    # celery to create its own app instance (which is not configured properly and is bugging the shared tasks).
+    # Also it must be done early because as soon as config scan is started, some packages may include celery and
+    # and it will create its own app instance.
+    if config.registry.settings.get(USE_PYRAMID_CELERY_APP_CFG, True):
+        pyramid_celery_app.set_default()
+
+    # Add the config dir in path so that celeryconfig file can be found
+    sys.path.append(os.path.dirname(config_ini))
+    config.include("pyramid_celery")
+    config.configure_celery(config_ini)
+
+    logger.info("Locating celery tasks...")
+    grep_command = ["grep", "--include=*.py", "-rw", os.path.dirname(__file__), "-e", "^@shared_task"]
+    task_files = subprocess.run(grep_command, stdout=subprocess.PIPE, text=True, check=True)  # nosec B603
```

Comment:
Is this related to problem to discover tasks via `pyramid_celery` include?
Maybe the issue is specific to `shared_task`?

Reply:
I feel there are is strong possibility `shared_task` is the cause. 
According to this, it should be used only in "utils" packages that over tasks used by other apps.
https://stackoverflow.com/questions/54506515/difference-between-different-ways-to-create-celery-task

In this case, cowbird is the app running the task, so `app.task` should be used as in weaver case.

Reply:
> Is this related to problem to discover tasks via pyramid_celery include?

Yes this is the way tasks are discovered and imported.

> I feel there are is strong possibility shared_task is the cause.

@shared_task is used because in tests, the app is not the same as in pyramid. 
Using celery built-in test fixture is based on this post : https://medium.com/@scythargon/how-to-use-celery-pytest-fixtures-for-celery-intergration-testing-6d61c91775d9

Reply:
This blog seems to make it extremely complicated for no reason.
I simply mock the execution of the celery task by calling it directly: 
https://github.com/crim-ca/weaver/blob/master/tests/utils.py#L425-L464

Unless you're actually running functional tests with both running instances of webapp/worker, I don't see the advantage of explicitly having celery running to test basic code functionality.

Reply:
In fact I heavily check celery. I want to test the chaining process, , the chaining interruption in case that a request cannot be recovered and the backoff strategy when a request cannot be completed at a time.
FYI : https://github.com/Ouranosinc/cowbird/blob/master/tests/test_request_task.py

Reply:
Ok. If the current approach works, that's fine. Tests must adapt to code and not the other way around.